feat: Add Recently active accounts section

Pytal opened the pull request:
## Summary

- Requires https://github.com/nextcloud/server/pull/45249
- Supersede https://github.com/nextcloud/server/pull/45500

| Screenshot |
| --- |
| ![image](https://github.com/nextcloud/server/assets/24800714/1a70ff11-4367-4863-9c8d-e4f290571786) |

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

artonge commented on apps/settings/lib/Controller/UsersController.php:
```diff
@@ -164,9 +164,15 @@ public function usersList(): TemplateResponse {
 
+		$recentUsersGroup = [
+			'id' => 'recent',
+			'name' => $this->l10n->t('Recent accounts'),
```
Not sure that it reflect the recently logged in logic. It sounds like recently created. Maybe @nextcloud/designers can give some input here.

susnux replied:
I think the problem is that we call it "active" when we mean "enabled".
Because then we could name it:
* Enabled -> all enabled ones
* Disabled -> all disabled (like it is)
* Active -> recently logged in as they are "active"

(not 100% sure that naming makes that much sense in English like it makes in German :sweat_smile: )

jancborchardt replied:
How about:
- All accounts
- Admins
- Recently active
- Disabled accounts

This sounds most like understandable language to me?

susnux commented:
> BREAKING CHANGE: The "recent" group id is now reserved for the Recent accounts section and will not be shown under the Groups heading

BTW could we maybe rename that to be less likly to conflict? Like `__nc_internal_recent`?

Pytal committed:
feat: Add Recent accounts section

Signed-off-by: Christopher Ng <chrng8@gmail.com>

Pytal committed:
chore: compile assets

Signed-off-by: Christopher Ng <chrng8@gmail.com>

Pytal committed:
chore(l10n): Translate group names

Signed-off-by: Christopher Ng <chrng8@gmail.com>

Pytal committed:
chore(settings): Rename user groups
- "Active accounts" -> "All accounts"
- "Recent accounts" -> "Recently active"

Signed-off-by: Christopher Ng <chrng8@gmail.com>
